### ISSUE TITLE
react-hooks/exhaustive-deps reports a local function that captures no reactive values

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1233]
## Summary

A local function with no captured values is reported as a missing dependency. ESLint treats this function as stable. Changing the function to capture a prop makes both tools correctly report the missing dependency.

Verified with @utoo/lint 0.4.1, ESLint 8.57.1, @typescript-eslint/parser 6.21.0, eslint-plugin-react-hooks 4.6.2, Node.js 22.23.1 on macOS arm64.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 eslint-plugin-react-hooks@4.6.2
```

`utlint.config.json`:

```json
{
  "rules": {
    "react-hooks/exhaustive-deps": "warn"
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react-hooks"
  ],
  "rules": {
    "react-hooks/exhaustive-deps": "warn"
  }
};
```

Run each example separately using:

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Reproduction 1

`example.tsx`:

```tsx
import { useMemo } from 'react';
export function Example(){ function getValue(){return 1;}const result=useMemo(()=>getValue(),[]);return result; }
```

```text
ESLint:
(no diagnostics)

utlint:
React Hook useMemo has a missing dependency: 'getValue'. Either include it or remove the dependency array.
```

## Controls

`example.tsx`:

```tsx
import {useMemo} from 'react'; export function Example({value}){function getValue(){return value;}return useMemo(()=>getValue(),[]);}
```

```text
ESLint:
React Hook useMemo has a missing dependency: 'getValue'. Either include it or remove the dependency array.

utlint:
React Hook useMemo has a missing dependency: 'getValue'. Either include it or remove the dependency array.
```

All examples parse successfully. No lint-disable comments or additional rules are involved.
